### PR TITLE
[docs] Add docs for reworked Fivetran API

### PR DESCRIPTION
**Pull request by maximearmstrong**
## Summary & Motivation

As title

## How I Tested These Changes

See preview

### CODE REVIEW

**Comment by maximearmstrong** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26026?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#26026** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26026?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26026?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#26025** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26025?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#26009** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/26009?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25961** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25961?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25944** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25944?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25911** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25911?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25889** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25889?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25808** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25808?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25807** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25807?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25797** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25797?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25788** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25788?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25756** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25756?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25751** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25751?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#25750** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/25750?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: examples/docs_snippets/docs_snippets/integrations/fivetran/customize-fivetran-asset-defs.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    workspace=fivetran_workspace,
+    dagster_fivetran_translator=MyCustomFivetranTranslator(),
+)
+def fivetran_connector_assets(
+    context: dg.AssetExecutionContext, fivetran: FivetranWorkspace
+):
+    yield from fivetran.sync_and_poll(context=context)
+
+
+# Alternatively, when creating all assets definitions for the Fivetran workspace
```
Comment (dpeng817) [inline examples/docs_snippets/docs_snippets/integrations/fivetran/customize-fivetran-asset-defs.py:52]: grammar here seems a bit off

File: examples/docs_snippets/docs_snippets/integrations/fivetran/representing-fivetran-assets.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+fivetran_workspace = FivetranWorkspace(
+    account_id=dg.EnvVar("FIVETRAN_ACCOUNT_ID"),
+    api_key=dg.EnvVar("FIVETRAN_API_KEY"),
+    api_secret=dg.EnvVar("FIVETRAN_API_SECRET"),
+)
+
+fivetran_specs = load_fivetran_asset_specs(fivetran_workspace)
+defs = dg.Definitions(
+    assets=[*fivetran_specs], resources={"fivetran": fivetran_workspace}
```
Comment (dpeng817) [inline examples/docs_snippets/docs_snippets/integrations/fivetran/representing-fivetran-assets.py:13]: should be able to just fivetran_specs, no need for splat operator right

File: examples/docs_snippets/docs_snippets/integrations/fivetran/sync-and-materialize-fivetran-assets.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import dagster as dg
+
+fivetran_workspace = FivetranWorkspace(
+    account_id=dg.EnvVar("FIVETRAN_ACCOUNT_ID"),
+    api_key=dg.EnvVar("FIVETRAN_API_KEY"),
+    api_secret=dg.EnvVar("FIVETRAN_API_SECRET"),
+)
+
+
+# Creating assets definition for a given connector using the `@fivetran_assets` decorator
```
Comment (dpeng817) [inline examples/docs_snippets/docs_snippets/integrations/fivetran/sync-and-materialize-fivetran-assets.py:16]: I think this comment is a bit redundant

**Review by dpeng817 — CHANGES_REQUESTED** [review]
Python files have hyphenated names, let's fix that.

Additionally, I think the structuring here is a bit weird. We should probably lead with `build_fivetran_assets_...` since that's the most straightforward way. Then, you can talk about "customizing execution functionality" and go through using the other APIs.

A few additional comments as well.

File: docs/content/integrations/fivetran/fivetran.mdx [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    assets=[*all_fivetran_assets],
+    resources={"fivetran": fivetran_workspace},
+)
+```
+
+### Customize asset definition metadata for Fivetran assets
+
+By default, Dagster will generate asset specs for each Fivetran asset and populate default metadata. You can further customize asset properties by passing a custom <PyObject module="dagster_fivetran" object="DagsterFivetranTranslator" /> subclass to the <PyObject module="dagster_fivetran" method="load_fivetran_asset_specs" /> function, the <PyObject module="dagster_fivetran" method="fivetran_assets" /> decorator or the <PyObject module="dagster_fivetran" method="build_fivetran_assets_definitions" /> factory.
+
+```python file=/integrations/fivetran/customize-fivetran-asset-defs.py
```
Comment (dpeng817) [inline docs/content/integrations/fivetran/fivetran.mdx:99]: This whole code snippet feels really long, can we make use of `startbefore` and `endafter` to trim this down to only the relevant bits?

Comment (maximearmstrong) [reply docs/content/integrations/fivetran/fivetran.mdx]: I updated it in [0d36d97](https://github.com/dagster-io/dagster/pull/26026/commits/0d36d97fc555ba84a75e34bb0cf899eb1a49d30a). The code snippet only showcase the load_fivetran_asset_specs use case now, and I added a note after the example saying that passing a custom translator can also be done with the decorator and the factory.

File: docs/content/integrations/fivetran/fivetran.mdx [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+
+defs = dg.Definitions(
+    assets=all_fivetran_assets,
+    resources={"fivetran": fivetran_workspace},
+)
+```
+
+### Customize the materialization of Fivetran assets
+
+You can use the <PyObject module="dagster_fivetran" method="fivetran_assets" /> decorator to create the assets definition for a given connector. Using this approach, you can customize the code within your \`\`@fivetran_assets\`-decorated function.
```
[the remark this answers is not in the document]
Comment (maximearmstrong) [reply docs/content/integrations/fivetran/fivetran.mdx]: Fixed in [0f9ef78](https://github.com/dagster-io/dagster/pull/26026/commits/0f9ef789465274214e93491d8146cd68b3dad4eb)

File: docs/content/integrations/fivetran/fivetran.mdx [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+
+defs = dg.Definitions(
+    assets=all_fivetran_assets,
+    resources={"fivetran": fivetran_workspace},
+)
+```
+
+### Customize the materialization of Fivetran assets
+
+You can use the <PyObject module="dagster_fivetran" method="fivetran_assets" /> decorator to create the assets definition for a given connector. Using this approach, you can customize the code within your `@fivetran_assets`-decorated function.
```
Comment (dpeng817) [inline docs/content/integrations/fivetran/fivetran.mdx:79]: Wording here feels a bit backward. Maybe:

If you want to customize the sync of your connectors, you can use the fivetran_assets decorator to do so. This allows you to execute custom code before and after the call to the fivetran sync.

File: docs/content/integrations/fivetran/fivetran.mdx [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+@fivetran_assets(
+    connector_id="fivetran_connector_id",
+    name="fivetran_connector_id",
+    group_name="fivetran_connector_id",
+    workspace=fivetran_workspace,
+)
+def fivetran_connector_assets(
+    context: dg.AssetExecutionContext, fivetran: FivetranWorkspace
+):
+    # Do something before the materialization...
```
Comment (dpeng817) [inline docs/content/integrations/fivetran/fivetran.mdx:106]: I would like something more concrete here. Like, what is a realistic scenario where someone would want to customize this?

File: docs/content/integrations/fivetran/fivetran.mdx [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+import dagster as dg
+
+fivetran_workspace = FivetranWorkspace(
+    account_id=dg.EnvVar("FIVETRAN_ACCOUNT_ID"),
+    api_key=dg.EnvVar("FIVETRAN_API_KEY"),
+    api_secret=dg.EnvVar("FIVETRAN_API_SECRET"),
+)
+
+
+# A translator class lets us customize properties of the built
```
Comment (dpeng817) [inline docs/content/integrations/fivetran/fivetran.mdx:137]: I thought we agreed on using replace_attributes / map_asset_specs for this?

File: docs/content/integrations/fivetran/fivetran.mdx [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+fivetran_specs = load_fivetran_asset_specs(
+    fivetran_workspace, dagster_fivetran_translator=MyCustomFivetranTranslator()
+)
+
+defs = dg.Definitions(assets=fivetran_specs, resources={"fivetran": fivetran_workspace})
+```
+
+Note that `super()` is called in each of the overridden methods to generate the default asset spec. It is best practice to generate the default asset spec before customizing it.
+
+Alternatively, you can also pass an instance of the custom <PyObject module="dagster_fivetran" object="DagsterFivetranTranslator" /> to the <PyObject module="dagster_fivetran" method="fivetran_assets" /> decorator or the <PyObject module="dagster_fivetran" method="build_fivetran_assets_definitions" /> factory.
```
Comment (dpeng817) [inline docs/content/integrations/fivetran/fivetran.mdx:159]: Delete alternatively here - Should just be "you can pass ...."

**Review by dpeng817 — APPROVED** [review]
Accepting but still a good number of comments. Heed the wording stuff before landing pls, but more structural stuff can be done in follow-ups

File: docs/content/integrations/fivetran/fivetran.mdx (hunk 8 above)
Comment (maximearmstrong) [reply docs/content/integrations/fivetran/fivetran.mdx:137]: That makes sense - will add that in a separate PR

File: docs/content/integrations/fivetran/fivetran.mdx (hunk 7 above)
Comment (maximearmstrong) [reply docs/content/integrations/fivetran/fivetran.mdx:98]: The best would be to add more examples, like when using configs and partitions. I can add an example with configs in another PR. About partitions, they are not supported yet with Fivetran.

**Pull request merged**